### PR TITLE
fatch data of bootcamp add bootcamp and dashboard

### PR DESCRIPTION
**Pull request by JayPagada**

### CODE REVIEW

File: src/Store/Action/Bootcamp.jsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        dispatch(getCurrentBootcampFailure(error.message))
+      })
+  };
+};
+export const deleteData = (id)=>{
+  const token = localStorage.getItem('token');
+  return dispatch => {
+    dispatch(getCurrentBootcampRequest());
+    api.delete('bootcamps/{bootcampId}',{
+      headers:{'Authorization':`Bearer ${token}`},
```
Comment (HirenTariwala) [inline src/Store/Action/Bootcamp.jsx:43]: Check Axios and make one global object of Axios there you can add this header we don't have to this in each and every request

File: src/Store/Reducer/Bootcamp.jsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+const reducer = (state = initialState, action) => {
+  switch (action.type) {
+    case actionTypes.GET_CURRENT_BOOTCAMP_REQUEST:
+      return {
+        ...state,
+        loading: true
+      }
+    case actionTypes.GET_CURRENT_BOOTCAMP_SUCCESS:
+      const bootCamps = action.bootCampsData.map((bootcamp, index) => {
```
Comment (HirenTariwala) [inline src/Store/Reducer/Bootcamp.jsx]: Add nested object checker action?.bootCampsData? like this

File: src/App.js [hunk 3]
```diff
@@ -6,22 +6,24 @@ import {Route, Redirect , Switch} from "react-router-dom"
 import Register from "./Containers/Register/Register";
 import MainLayout from "./Components/Layout/MainLayout";
 import * as actions from "./Store/Action/Login.jsx";
-import {useDispatch,connect} from "react-redux";
+import {useDispatch,useSelector} from "react-redux";
 
 function App(props) {
   const dispatch = useDispatch()
+  const auth = useSelector(state => state);
```
Comment (webdevlapani) [inline src/App.js:14]: You need to extract a particular state like `const auth = useSelector(state => state.Like);` insted of whole state.
and `Login` should start with small `login`. You can use camel-case that's fine

File: src/App.js [hunk 4]
```diff
@@ -6,22 +6,24 @@ import {Route, Redirect , Switch} from "react-router-dom"
...
   let routes = (
     <Switch>
       <Route path="/Login" exact component={Login}/>,
       <Route path="/Register" exact component={Register}/>,
       <Redirect to="/" from="/Login"/>
     </Switch>
   );
-  if (props.isAuthenticated) {
+
+  if (auth.Login.token !== null) {
```
Comment (webdevlapani) [inline src/App.js]: You don't need to check like this `if (auth.Login.token !== null) `.
You can check condition like this `if (auth.Login.token) `

File: src/Store/Action/Bootcamp.jsx (hunk 1 above)
Comment (JayPagada) [reply src/Store/Action/Bootcamp.jsx:43]: I'm trying to work on it

File: src/App.js [hunk 5]
```diff
@@ -10,7 +10,7 @@ import {useDispatch,useSelector} from "react-redux";
 
 function App(props) {
   const dispatch = useDispatch()
-  const auth = useSelector(state => state);
+  const auth = useSelector(state => state.Login);
```
Comment (HirenTariwala) [inline src/App.js:13]: object name var name always in camelCase

**Pull request merged**